framework/src/main/java/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -546,15 +577,29 @@ private AnnotationMirror createAnnotationForStringConcatenation(
         List<String> leftValues = getStringValues(leftOperand, p);
         List<String> rightValues = getStringValues(rightOperand, p);
 
+        boolean allowNullStringConcat =
+                atypefactory.getContext().getChecker().hasOption("nullStringsConcatenation");
+
         if (leftValues != null && rightValues != null) {
             // Both operands have known string values, compute set of results
             List<String> concatValues = new ArrayList<>();
+
+            if (allowNullStringConcat) {
+                if (isNullable(leftOperand)) {
+                    leftValues.add("null");
+                }
+                if (isNullable(rightOperand)) {
+                    rightValues.add("null");
+                }
+            }
+
             if (leftValues.isEmpty()) {
-                leftValues = Collections.singletonList("null");
+                leftValues.add("null");
```

Comment:
Why is this needed? Just to preserve the old behavior?
Seems to be an exception from "The Constant Value Checker considers all elements of a String concatenation non-null". If so, and we want to keep this behavior, could it at least be guarded by `isNullable` to reduce false positives?